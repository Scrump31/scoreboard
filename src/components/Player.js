import React from 'react';
import { Col } from 'react-bootstrap';
import Counter from './Counter';

export default function Player(props) {
  return (
    <div>
      <Col md={6} className="player-name"><div>{props.name}</div></Col>
      <Col md={6} className="player-counter">
        <Counter score={props.score} />
      </Col>
      <Col md={12} className="horizonal-rule" />
    </div>
  );
}
Player.propTypes = {
  name: React.PropTypes.string.isRequired,
  score: React.PropTypes.number,
}.isRequired;
