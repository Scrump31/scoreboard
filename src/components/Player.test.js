import React from 'react';
import { mount } from 'enzyme';
import Player from './Player';

const REMOVE_BTN = 'button.glyphicon-remove';
const PLAYER_NAME = 'div.player-name.col-xs-9 > div';

let component;
let mockDelPlayer;

beforeEach(() => {
  mockDelPlayer = jest.fn();
  component = mount(<Player name={'test name'} deletePlayer={mockDelPlayer} />);
});

afterEach(() => component.unmount());

describe('<Player />', () => {
  it('displays a name based on props.name', () => {
    expect(component.find(PLAYER_NAME).text()).toEqual('test name');
  });

  it('calls mockDelPlayer() when "x" button clicked', () => {
    component.find(REMOVE_BTN).simulate('click');
    expect(mockDelPlayer).toHaveBeenCalled();
  });
});
