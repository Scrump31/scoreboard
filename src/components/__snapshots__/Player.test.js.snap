// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders without crashing 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    className="player-name"
    componentClass="div"
    xs={9}
  >
    <div>
      test name
      <button
        className="glyphicon glyphicon-remove"
        onClick={[Function]}
      />
    </div>
  </Col>
  <Col
    bsClass="col"
    className="player-counter"
    componentClass="div"
    xs={3}
  >
    <Counter
      id={1}
      score={2}
      updateScore={[Function]}
    />
  </Col>
  <Col
    bsClass="col"
    className="horizonal-rule"
    componentClass="div"
    md={12}
  />
</Row>
`;
